lib/stages/BeamAssemble.hpp

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2022 Kotekan Developers
+
+/****************************************************
+ * @file   BeamAssemble.hpp
+ * @brief  This file declares the BeamAssemble class which
+ *         takes an input buffer with frames for single
+ *         frequency, and assembles the frames for a given
+ *         number of frequencies in a larger buffer.
+ *
+ * @author Mehdi Najafi
+ * @date   28 AUG 2022
+ *****************************************************/
+
+#ifndef KOTEKAN_BEAMASSEMBLE_HPP
+#define KOTEKAN_BEAMASSEMBLE_HPP
+
+#include "Config.hpp"          // for Config
+#include "Stage.hpp"           // for Stage
+#include "bufferContainer.hpp" // for bufferContainer
+
+#include <stdint.h>            // for uint64_t, int32_t, uint32_t
+#include <string>              // for string
+
+/**
+ * @class BeamAssemble
+ * @brief Assembles incoming beam information from the frames in
+ *        the in_buf and gather in a larger format to an out_buf.
+ * *
+ * @par Buffers
+ * @buffer in_buf The buffer to receive data from.
+ *     @buffer_format 4+4-bit complex voltage (beam) data
+ *     @buffer_metadata BeamMetadata
+ * @buffer out_buf The buffer to put the data to.
+ *     @buffer_format 4+4-bit complex voltage (beam) data
+ *     @buffer_metadata BeamMetadata
+ *
+ * @conf num_freq_per_output_frame
+ *      number of frequencies to be assembled
+ *
+ * @conf arriving_data_timeout
+ *      timeout in miliseconds for the arriving data
+ *
+ * @conf beam_printout
+ *      check flag to print some information about incoming beam
+ *      data
+ *
+ * @conf late_beam_printout
+ *      check flag to print some information about late incoming
+ *      beam data, which are ignored
+ *
+ * @par received_beam_frames_count
+ *      a counter to keep track of the number of beam frames received
+ *
+ * @par accepted_beam_frames_count
+ *      a counter to keep track of the number of beam frames received
+ *      and copied to the output buffer frames
+ *
+ * @par missed_beam_frames_count
+ *      a counter to keep track of the number of beam frames received
+ *      late and ignored (not copied to any output buffer frames)
```

Comment:
Since these are private members, and not exposed via the metrics API, I would just put these comments over the variables themselves in the class below.  That also will annotate them directly in Doxygen.

Reply:
I agree, shall I keep the above comments as well?

Reply:
I'd remove the comments them from the class header.